src/all/novelfull/src/eu/kanade/tachiyomi/extension/all/novelfull/NovelFullGlobal.kt

```diff
@@ -0,0 +1,256 @@
+
+    override fun mangaDetailsParse(document: Document): SManga {
+        return SManga.create().apply {
+            title = document.selectFirst("h3.title")!!.text()
+            description = document.selectFirst(".desc-text")!!.text()
+                .replace("\\n", "\n")
+            thumbnail_url = document.selectFirst(".books > .book > img")!!.imgAttr()
+            document.selectFirst(Evaluator.Class("info"))?.children()?.forEach {
+                when (it.children().first()?.text()) {
+                    // No need to translate these strings.
+                    "Author:" -> it.parseAuthorsTo(this)
+                    "Genre:" -> it.parseGenresTo(this)
+                    "Status:" -> it.parseStatusTo(this)
+                }
+            }
+        }
+    }
+
+    private fun Element.parseAuthorsTo(manga: SManga) {
+        manga.author = this.select("a").joinToString { it.text() }
+    }
+
+    private fun Element.parseGenresTo(manga: SManga) {
+        manga.genre = this.select("a").joinToString { it.text() }
+    }
+
+    private fun Element.parseStatusTo(manga: SManga) {
+        this.selectFirst("h3:last-child").let { element ->
+            manga.status = when (element?.text()?.lowercase()) {
+                // No need to translate these strings.
+                "ongoing" -> SManga.ONGOING
+                "completed" -> SManga.COMPLETED
+                else -> {
+                    Log.e("NovelFullGlobal", "Unknown status: ${element?.text()}")
+                    SManga.UNKNOWN
+                }
+            }
+        }
+    }
+
+    // =============================== Chapters ==============================
+
+    private fun chapterListRequest(manga: SManga, page: Int): Request {
+        val url = (baseUrl + manga.url).toHttpUrl().newBuilder()
+        if (page != 1) { // Cache ?
+            url.addQueryParameter("page_num", page.toString())
+        }
+        return GET(url.build(), headers)
+    }
+
+    override fun chapterListRequest(manga: SManga): Request {
+        throw UnsupportedOperationException("Not used")
+    }
+
+    override fun fetchChapterList(manga: SManga): Observable<List<SChapter>> {
+        // We need to get multiple pages for the chapters.
+        // The first page is already loaded in the manga details page.
+        // The rest of the pages are loaded with ?page_num=$page
+        // Chapter list will be empty when there are no more pages.
+        return client.newCall(chapterListRequest(manga, 1))
+            .asObservableSuccess()
+            .flatMap { firstResponse ->
+                val firstDocument = firstResponse.asJsoup()
+
+                // Determine the last page number from the pagination section.
+                val lastPageElement = firstDocument.selectFirst(".last a")
+                    ?.attr("href")
+
+                val lastPageNumber = baseUrl.toHttpUrl().newBuilder().build()
+                    .resolve(lastPageElement!!)
```

Comment:
What's the point of a non-null assert here _and_ a fallback value? Try accounting for this variable being null.